src/entity/projectile/SplashPotion.php

```diff
@@ -126,10 +127,22 @@ protected function onHit(ProjectileHitEvent $event) : void{
 						}
 					}
 				}
-			}else{
-				//TODO: lingering potions
 			}
-		}elseif($event instanceof ProjectileHitBlockEvent && $this->getPotionType() === PotionType::WATER){
+		}else{
+			$entity = new AreaEffectCloud(Location::fromObject($this->location->floor()->add(0.5, 0.5, 0.5), $this->getWorld()), $this->potionType);
+			$entity->setRadiusChangePerTick(-$entity->getInitialRadius() / $entity->getDuration());
+			if(($owner = $this->getOwningEntity()) !== null && !$owner->isClosed()){
+				$entity->setOwningEntity($owner);
+			}
+			foreach($this->getPotionEffects() as $effect){
+				$entity->getEffects()->add($effect);
+				if($effect->getType() instanceof InstantEffect){
+					$entity->setReapplicationDelay(0);
+				}
+			}
```

Comment:
I don't like this.

Currently we have an AEC entity that takes a PotionType in the constructor, but the potion type is useless because the entity doesn't use it. No effects are taken from the potion type. Good way to troll plugin devs trying to spawn AECs with a specific potion type.